### ISSUE TITLE
Cucumber beta3 doesn't work with RubyMine

### ISSUE THREAD

**Issue by luke-hill** [cucumber/cucumber-ruby#1028]
This is caused by a problem which RubyMine I assume believed was fixed in version 2.

The logic states that if you have cucumber version 2, you use the new Cucumber command `::Cucumber::Core::Ast::ScenarioOutline` and otherwise you use `::Cucumber::Ast::ScenarioOutline`

Naturally this needs modifying so that if you have cucumber 2 OR 3, you use the new command.

Problematic file/s are:

`C:/Program Files (x86)/JetBrains/RubyMine 2016.2.3/rb/testing/patch/bdd/teamcity/cucumber/common.rb:218:in`scenario_outline?'`
`C:/Program Files (x86)/JetBrains/RubyMine 2016.2.3/rb/testing/patch/bdd/teamcity/cucumber/common.rb:223:in `tc_before_feature_element'`
`C:/Program Files (x86)/JetBrains/RubyMine 2016.2.3/rb/testing/patch/bdd/teamcity/cucumber/formatter_03103.rb:98:in`before_feature_element'`

I'm not sure if this is something we can fix our end or if it's a RubyMine bug, as the file in question is an inbuilt one for RubyMine. Apologies if this is already known

Luke

**Comment by westlakem** [comment]
@luke-hill . I'm not on the project but I submitted an issue to Jetbrains on this issue.  Jetbrains fixed the issue in RubyMine2017.1 (currently in ESR, but I've verified it is fixed)

**Comment by luke-hill** [comment]
Will close once cucumber 3 and 2017.1 is released

**Comment by clergyman** [comment]
Hi everyone, I have the same problem with cucumber 3.0.0 + TeamCity 2017.1.4 (in fact, the problem erase starting from cucumber update, the TeamCity version is not important)
cucumber script exits after executing background for scenario with the following record in the log:
uninitialized constant Cucumber::Ast (NameError)
```
>>/Users/user/TeamCity/buildAgent2/plugins/rake-runner/rb/patch/bdd/teamcity/cucumber/common.rb:218:in `scenario_outline?'

>>/Users/user/TeamCity/buildAgent2/plugins/rake-runner/rb/patch/bdd/teamcity/cucumber/common.rb:223:in `tc_before_feature_element'

>>/Users/user/TeamCity/buildAgent2/plugins/rake-runner/rb/patch/bdd/teamcity/cucumber/formatter_03103.rb:98:in `before_feature_element'
```
Looks like a problem on team city side, everything works like charm when 
>> rake exec cucumber ...

**Comment by westlakem** [comment]
@clergyman yes, that is the same issue that I pointed at that is fixed in RubyMine 2017.1+

**Comment by luke-hill** [comment]
@aslakhellesoy (As I have tagged Matt a lot recently). What's the plans for this going forwards. Now 3.0.0 is out and a legit release. I have the latest RubyMine.

I'm aware this software isn't in your remit. But is there any way of prodding them? The solution is a hacky open `.rb` file as Admin and change lines of code
